controllers/database_controller.go

```diff
@@ -76,7 +74,24 @@ func (r *DatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	log.Info("Database CR Status: " + util.ToString(database.Status))
 
-	NDBInfo := database.Spec.NDB
+	// Fetch the NDBServer resource from the namespace
+	ndbServer := &ndbv1alpha1.NDBServer{}
+	ndbNamespacedName := req.NamespacedName
```

Comment:
Can you check if this does a deep/shallow copy

Reply:
Good catch, thanks!
Just checked, Golang copies the value internally. If this was a reference type (pointer etc) then it would have copied the reference only (essentially a shallow copy) but since we're directly assigning a struct/field it copies it. 

Here's a little exercise I did to verify this:
<img width="191" alt="image" src="https://github.com/nutanix-cloud-native/ndb-operator/assets/107612534/f0f989f5-2af4-4f4f-a02b-c2b1563ddf11">